From 35188911731b94d835d0110127e7097d9915a8d5 Mon Sep 17 00:00:00 2001
From: henry shao <hshao@netlogicmicro.com>
Date: Thu, 16 Jun 2011 15:02:51 -0700
Subject: [PATCH 233/565] bcm-xlp: add OHCI support.

Add OHCI support. Enable USB Keyboard and Mouse.
Add /dev/sdc and /dev/sdd to initramfs.

Based on Broadcom SDK 2.3.

Signed-off-by: henry shao <hshao@netlogicmicro.com>
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 arch/mips/netlogic/xlp/platform.c | 18 +++++++++++-------
 1 file changed, 11 insertions(+), 7 deletions(-)

diff --git a/arch/mips/netlogic/xlp/platform.c b/arch/mips/netlogic/xlp/platform.c
index a827659..c3db12e 100644
--- a/arch/mips/netlogic/xlp/platform.c
+++ b/arch/mips/netlogic/xlp/platform.c
@@ -109,23 +109,27 @@ static void xlp_init_uart(int port_id)
 static void xlp_usb_hw_start(int ctrl_no)
 {
 	int val;
+	
+	/* Disable 64bit enable
+ 	 */
+	val = usb_reg_read( 0, ctrl_no, XLP_USB_CTL0);
+	val &= ~USBEHCI64BITEN;
+	usb_reg_write(0, ctrl_no, XLP_USB_CTL0, val);
 
-	/* reset USB phy 
+	/* Reset USB phy 
 	 */
 	val = usb_reg_read( 0, ctrl_no, XLP_USB_PHY0);
 
 	if(ctrl_no == 0)
 		val &= ~(USBPHYRESET | USBPHYPORTRESET0 | USBPHYPORTRESET1);
 	else if(ctrl_no == 3)
-		val &= ~(USBPHYRESET | USBPHYPORTRESET0 | USBPHYPORTRESET1);
-
+		val &= ~(USBPHYPORTRESET0 | USBPHYPORTRESET1);
 	usb_reg_write(0, ctrl_no, XLP_USB_PHY0, val);
-
-	udelay(2000);
-
+	
+	/* Bring usb controller out of reset
+ 	 */
 	val = usb_reg_read( 0, ctrl_no, XLP_USB_CTL0);
 	val &= ~(USBCONTROLLERRESET );
-	val |= 0x4;
 	usb_reg_write(0, ctrl_no, XLP_USB_CTL0, val);
 
 	return;
-- 
1.8.4.93.g57e4c17

